### ISSUE TITLE
False positive in redos detection when backtracking doesn't occur

### ISSUE THREAD

**Issue by mschwager** [duo-labs/dlint#47]
In order for catastrophic backtracking to occur there must be a character that forces backtracking to occur. E.g.

```python
>>> re.match('(a+)+', 'a' * 64 + 'c')
<_sre.SRE_Match at 0x7f9106d32430>
>>> re.match('(a+)+b', 'a' * 64 + 'c')
...Spins...
```

The first expression has nested quantifiers, so Dlint will detect it, but since there's nothing after the nested quantifier to backtrack the match no catastrophic backtracking will occur. Similarly:

```
$ python -m dlint.redos -p '(a+)+'
('(a+)+', True)
$ python -m dlint.redos -p '(a+)+b'
('(a+)+b', True)
```

We should avoid detecting the first expression because catastrophic backtracking cannot occur.
